src/server/analytic_services/prometheus_reports/noobaa_endpoint_report.js

```diff
@@ -0,0 +1,44 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const { BasePrometheusReport } = require('./base_prometheus_report');
+
+// -----------------------------------------
+// A report for collecting endpoint metrics,
+// this class is a shim that should who ever
+// come next to add endpoint metrics reporting.
+// -----------------------------------------
+class NooBaaEndpointReport extends BasePrometheusReport {
+    constructor() {
+        super();
+        // Cache the client for ease of use.
+        const client = this.prom_client;
+
+        // Define storage for the metrics:
+        this._metrics = {
+            // This is a dummy metric that is used as an example, Gauge is use
+            // here forbut it can be any of the metric types defined in the prom
+            // client lib at https://github.com/siimon/prom-client with its
+            // specific configuration.
+            demo_metric: new client.Gauge({
```

Comment:
I think that this is more complex and prone to errors, rather than using the JSON configuration as we do in the NooBaa Core metrics.
This is just writing inline everything including object creation, instead of configuring that in a JSON and letting a dynamic code to configure all of that.

Reply:
@jeniawhite It is almost exactly the same text, you don't gain anything by removing it outside you are just making it harder to mentally associate the metrics.

It also comes with a baggage of cons, please see the discussion above with paul regarding this issue